pkg/cortex/serve/init/install-core-dependencies.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 Cortex Labs, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+function module_exists() {
+    module_name=$1
+    python -c "import importlib, sys; loader = importlib.util.find_spec('$module_name'); sys.exit(1) if loader is None else sys.exit(0)"
+}
+
+temp_freeze=$(mktemp)
+pip freeze > "${temp_freeze}"
+
+if ! grep -q "cortex-internal" "${temp_freeze}"; then
+    pip install --no-cache-dir -U \
```

Comment:
why the `-U` flag? Couldn't this cause problems if a dependency has breaking changes?

Reply:
If the dependencies' versions of `cortex-internal` don't match with those that already are in the environment, then upgrade them to the right versions. We need to ensure that the `cortex-internal` package is running on the right set of dependencies. If it weren't for that `-U` flag, then there could be a version of `uvicorn` that's far too old for `cortex-internal` to work correctly.

That being said, this is not ideal, because we don't have ranges associated with the dependencies. This is something that needs to be addressed in a subsequent PR.